shared/fs/header/header-container.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import * as Types from '../../constants/types/fs'
+import {compose, connect, setDisplayName, type Dispatch} from '../../util/container'
+import {fsTab} from '../../constants/tabs'
+import {navigateAppend, navigateTo} from '../../actions/route-tree'
+
+type OwnProps = {
+  path: Types.Path,
+}
+
+const mapStateToProps = () => ({})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  onOpenBreadcrumb: (path: string) =>
+    dispatch(navigateTo([fsTab, {props: {path: Types.stringToPath(path)}, selected: 'folder'}])),
+  onOpenBreadcrumbDropdown: (
+    dropdownItems: Array<Types.PathBreadcrumbItem>,
+    isTeamPath: boolean,
+    onOpenBreadcrumb: (path: string) => void,
+    targetRect: ?ClientRect
+  ) =>
+    dispatch(
+      navigateAppend([
+        {
+          props: {
+            targetRect,
+            position: 'top right',
+            isTeamPath,
+            items: dropdownItems,
+            onOpenBreadcrumb,
+          },
+          selected: 'breadcrumbAction',
+        },
+      ])
+    ),
+})
+
+const mergeProps = (stateProps, dispatchProps, {path}: OwnProps) => {
+  let acc = Types.stringToPath('/')
+  const elems = Types.getPathElements(path)
+  const items = elems.map((e, i) => {
+    acc = Types.pathConcat(acc, e)
+    return {
+      idx: i,
```

Comment:
the other option is to just set this as isTeamItem and isLastItem so you don't have to have additional idx logic in the dumb component